### PR TITLE
docs(configuration): reorder sections, improve wording, add references in concepts

### PR DESCRIPTION
**Pull request by byzyk**
* Moved `Use custom config file` section to the bottom since it's less relevant than actual options instead. 
* Added config reference in concepts
* Fixed wording in some places
* Minor adjustments to bring everything to common formatting in MD (perhaps we could add this to MD linter?)

### CODE REVIEW

File: src/content/concepts/configuration.md [hunk 1]
```diff
@@ -45,8 +46,12 @@ module.exports = {
 };
 ```
 
+_See_: [Configuration section](/configuration) for the all supported config options
+
 ## Multiple Targets
 
+Along with exporting a single configuration object/function, webpack supports exporting multiple configurations.
```
Comment (EugeneHlushko) [inline src/content/concepts/configuration.md]: __webpack supports exporting multiple configurations__ - this line makes me think that webpack will export the configurations, when that is user's responsibility to write a configuration module with export statement.

suggest rewording into: webpack can accept multiple configurations

File: src/content/configuration/index.md [hunk 2]
```diff
@@ -396,9 +383,23 @@ module.exports = {
 }
 ```
 
-T> Want to rapidly generate webpack configuration file for your project requirements with few clicks away.
+## Use custom config file
+
+If for some reason you want to use custom config file depending on certain situations you can change this via command line by using the `--config` flag.
```
Comment (EugeneHlushko) [inline src/content/configuration/index.md]: i understand why title was shortened, but in the text body suggest to change `config file` -> `configuration file`

File: src/content/configuration/index.md [hunk 3]
```diff
@@ -396,9 +383,23 @@ module.exports = {
...
+
+__package.json__
+
+```json
+"scripts": {
+  "build": "webpack --config prod.config.js"
+}
+```
+
+## Config generators
```
Comment (EugeneHlushko) [inline src/content/configuration/index.md]: maybe `config` -> `configuration` in this title sits well in left hand side menu?

File: src/content/concepts/configuration.md (hunk 1 above)
Comment (byzyk) [reply src/content/concepts/configuration.md]: Fair enough 👍🏽

File: src/content/configuration/index.md (hunk 2 above)
Comment (byzyk) [reply src/content/configuration/index.md]: That's the thing, we've got both `config` and `configuration` all over the docs and sometimes it's hard to justify why one was picked over another one. I don't wanna seem too picky here but perhaps this can be brought to some common option? Also not sure if that's worth it

Comment (montogeek) [reply src/content/configuration/index.md]: @byzyk Yep, we have both, but we can start with this opportunity to change it to Configuration

File: src/content/concepts/configuration.md [hunk 4]
```diff
@@ -4,25 +4,26 @@ sort: 6
...
 You may have noticed that few webpack configurations look exactly alike. This is because **webpack's configuration file is a JavaScript file that exports an object.** This object is then processed by webpack based upon its defined properties.
 
 Because it's a standard Node.js CommonJS module, you **can do the following**:
 
-* import other files via `require(...)`
-* use utilities on npm via `require(...)`
-* use JavaScript control flow expressions i. e. the `?:` operator
-* use constants or variables for often used values
-* write and execute functions to generate a part of the configuration
+- import other files via `require(...)`
```
Comment (montogeek) [inline src/content/concepts/configuration.md]: Please use title case in lists, same as #L24

Comment (byzyk) [reply src/content/concepts/configuration.md]: oops, seems like prettier formatted it automatically. will update tom

**Comment by byzyk** [comment]
hey @montogeek, this one is ready to go

**Comment by byzyk** [comment]
Hey @montogeek, I've resolved conflicts yesterday and now it seems ok. Do you have anything else on your mind to add here before we proceed?

File: src/content/configuration/index.md [hunk 5]
```diff
@@ -23,25 +24,11 @@ All the available configuration options are specified below.
...
-}
-```
-
-
 ## Options
 
 Click on the name of each option in the configuration code below to jump to the detailed documentation. Also note that the items with arrows can be expanded to show more examples and, in some cases, more advanced configuration.
 
-W> Notice that throughout the configuration we use Node's built-in [path module](https://nodejs.org/api/path.html) and prefix it with the [__dirname](https://nodejs.org/docs/latest/api/globals.html#globals_dirname) global. This prevents file path issues between operating systems and allows relative paths to work as expected. See [this section](https://nodejs.org/api/path.html#path_windows_vs_posix) for more info on POSIX vs. Windows paths.
+W> Notice that throughout the configuration we use Node's built-in [path module](https://nodejs.org/api/path.html) and prefix it with the [\_\_dirname](https://nodejs.org/docs/latest/api/globals.html#globals_dirname) global. This prevents file path issues between operating systems and allows relative paths to work as expected. See [this section](https://nodejs.org/api/path.html#path_windows_vs_posix) for more info on POSIX vs. Windows paths.
```
Comment (EugeneHlushko) [inline src/content/configuration/index.md]: why do we need that change ? `[__dirname] => [\_\_dirname]`

Comment (byzyk) [reply src/content/configuration/index.md]: I guess prettier changed it accidentally, will revert later today

Comment (EugeneHlushko) [reply src/content/configuration/index.md]: ok, i think its good to go, fernando doesnt have time to review this so ping me

**Comment by byzyk** [comment]
yo @EugeneHlushko, please take a look 🙂

File: src/content/concepts/configuration.md [hunk 6]
```diff
@@ -46,8 +47,12 @@ module.exports = {
 };
 ```
 
+_See_: [Configuration section](/configuration) for the all supported configuration options
```
Comment (EugeneHlushko) [inline src/content/concepts/configuration.md]: can you please change to `/configuration/` so it also works on dev ? dev server doesn't work for non trailing slash urls

File: src/content/configuration/index.md [hunk 7]
```diff
@@ -62,7 +49,7 @@ module.exports = {
     b: ["./app/entry-b1", "./app/entry-b2"]
   },
   </details>
-  // defaults to ./src
+  // defaults to "./src"
```
Comment (EugeneHlushko) [inline src/content/configuration/index.md]: single quotes

File: src/content/concepts/configuration.md [hunk 8]
```diff
@@ -5,11 +5,12 @@ contributors:
   - TheLarkInn
   - simon04
   - EugeneHlushko
+  - byzyk
 ---
 
-You may have noticed that few webpack configurations look exactly alike. This is because **webpack's configuration file is a JavaScript file that exports an object.** This object is then processed by webpack based upon its defined properties.
+You may have noticed that few webpack configurations look exactly alike. This is because __webpack's configuration file is a JavaScript file that exports an object.__ This object is then processed by webpack based upon its defined properties.
```
Comment (EugeneHlushko) [inline src/content/concepts/configuration.md]: later on we mention that configuration is an object or a function please also mention it here in the `__` section

Comment (byzyk) [reply src/content/concepts/configuration.md]: I don't think it's necessary to mention it as it's covered in more details [here](https://webpack.js.org/configuration/configuration-types/#exporting-multiple-configurations). You can export a function, Promise or array but in the end, it's just a plain object returned. 

Maybe we can add it as a note below, saying something like `webpack configuration is always presented as an object but can be exported as an array, promise, function etc + include the link I shared above`. What do you think?

Comment (EugeneHlushko) [reply src/content/concepts/configuration.md]: Maybe change it to `... that exports a webpack [configuration](/configuration/)` because `javascript file that exports ....` is not about how its presented but about what it exports explicitly

Comment (byzyk) [reply src/content/concepts/configuration.md]: I guess the idea here was to show that webpack config is just a plain JS object and you can mess up with it in any way you like. Changing it to `exports webpack config` might lead to confusion, especially among beginners.

Comment (byzyk) [reply src/content/concepts/configuration.md]: Then there is more advanced guide on how to export function or array etc. and we can add that one as a note here, this will allow us to keep the existing guide more beginner friendly and have a reference that exporting an object is not the only option. Does it make sense?

Comment (EugeneHlushko) [reply src/content/concepts/configuration.md]: dunno man its plain straight to me that we can export w/e from javascript webpack config file, lets hear from more folks then

CC @jeremenichelli @montogeek

Comment (EugeneHlushko) [reply src/content/concepts/configuration.md]: on the other hand, the text was like this before (its not like you've updated it, just the syntax around it). we can change it later if more folks agree on it

**Comment by byzyk** [comment]
@EugeneHlushko please see my last commit, is that something you had on your mind?

**Comment by EugeneHlushko** [comment]
yup, lets wait for tests and LGTM

**Comment by EugeneHlushko** [comment]
Thanks!
